contracts/governance/Staking/Staking.sol

```diff
@@ -774,4 +957,34 @@ contract Staking is
                 block.number;
         require(notSameBlock, "S20"); //S20 : "cannot be mined in the same block as last stake"
     }
+
+    /**
+     * @notice Max iteration for direct withdrawal from staking to prevent out of gas issue.
+     *
+     * @return max iteration value.
+     */
+    function getMaxVestingWithdrawIterations() public view returns (uint256) {
+        return maxVestingWithdrawIterations;
+    }
+
+    /**
+     * @dev set max withdraw iterations.
+     *
+     * @param maxIterations new max iterations value.
+     */
+    function setMaxVestingWithdrawIterations(uint256 maxIterations) external onlyAuthorized {
+        require(maxIterations > 0, "S21");
+        uint256 oldIterations = maxVestingWithdrawIterations;
+        maxVestingWithdrawIterations = maxIterations;
+
+        emit MaxVestingWithdrawIterationsUpdated(oldIterations, maxVestingWithdrawIterations);
+    }
+
+    function setVestingCreationType(uint256 _duration, uint256 _vestingCreationType)
+        external
+        onlyAuthorized
+    {
+        vestingCreationType[_duration] = _vestingCreationType;
+        emit VestingCreationTypeUpdated(_duration, _vestingCreationType);
+    }
```

Comment:
thsi really shouldn't be here. read teh long comment which summarizes the review for more details.

Reply:
If we create a new mapping in the vestingRegistryLogic, I assumed the value will be set everytime we createVesting & createTeamVesting?

But how are we dealing with the previous created vesting contracts? I guess we need to create some sort of setter function for the existing one?

Reply:
yes. we would need a batch setter function. i still think this is much cleaner than adding vesting types to the staking contract. we'll also need a script for it.

Reply:
updated, and added the script as well